expanalysis/results.py

```diff
@@ -0,0 +1,166 @@
+# -*- coding: utf-8 -*-
+"""
+Created on Thu Mar 24 16:03:04 2016
+
+@author: ian
+"""
+
+from expanalysis.api import get_results
+from utils import clean_df
+import pandas
+
+class Results:
+    def __init__(self, access_token, clean = True):
+        self.data_json = get_results(access_token=access_token) #original data
+        self.data = self.results_to_df() #active data
+        if clean:
+            self.clean_results()
+        self.battery = None
+        self.experiment = None
+        self.worker = None
+    
+    #*******************************************
+    #
+    #*******************************************
+    def set_battery(self, battery, reset = False):
+        '''Subset data to the specific battery, or array of batteries
+            battery may be an array or a string. If reset is true, the data will
+            be reset to a cleaned dataframe
+        '''
+        if reset:
+            self.reset_data()
+        self.battery = battery
+        self.data = self.select_battery(battery)
+        
+    def results_to_df(self):
+        if isinstance(self.data_json,list):
```

Comment:
take a look at the other modules to see how to document the functions, eg with :param variable: and a description - this is important for when we have Sphinx docs, so the function details get parsed.